control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -335,15 +335,13 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return nil
 	}
 
-	_, externalTopic := isExternalTopic(broker)
 	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig, UseNamespaceInConfigmap: false}, r.SecretProviderFunc())
 	if err != nil {
-
-		if externalTopic {
-			// we don't care
-			return nil
-		} else {
-			return fmt.Errorf("failed to get secret: %w", err)
+		// If we can not get the referenced secret,
+		// let us try for a bit before we give up.
+		brokerUUID := string(broker.GetUID())
+		if r.Counter.Inc(brokerUUID) <= 5 {
+			return controller.NewRequeueAfter(5 * time.Second)
 		}
```

Comment:
by continuing without returning what happens when we call `securityOption := security.NewSaramaSecurityOptionFromSecret(secret)` with a `nil` secret?

Reply:
```
	if secret == nil {
		return kafka.NoOpConfigOption
	}
```